web/src/overlays.ts

```diff
@@ -27,6 +27,9 @@ export type ModalConfig = {
 let active_overlay: Overlay | undefined;
 let open_overlay_name: string | undefined;
 
+// Used for both overlays and modals.
+// We could split these hooks so that they are separate for
+// overlays and modals if we need to.
 const pre_open_hooks: Hook[] = [];
 const pre_close_hooks: Hook[] = [];
```

Comment:
Yeah, let's plan to do that -- I think actually the best plan is to split out of this module a `modals.ts` -- there's no reason that modals are in `overlays.ts` and it's pretty confusing, and each one can have its own copy of these open/close hooks when we do the split, so we'd have `popover_menus.js` registering hooks for both systems.